internal/repository/repository.go

```
@@ -0,0 +1,638 @@
+	if err := Preflight(ctx, []Target{target}, runner); err != nil {
+		return Receipt{}, err
+	}
+	return createPrepared(ctx, target, runner)
+}
+
+func createPrepared(ctx context.Context, target Target, runner Runner) (Receipt, error) {
+	workspace, err := os.MkdirTemp("", "agx-repository-")
+	if err != nil {
+		return Receipt{}, fmt.Errorf("AGX-REPOSITORY-STAGING: cannot create temporary directory: %w", err)
+	}
+	defer os.RemoveAll(workspace)
```

golangci-lint (2.12.2)

[error] Error return value of `os.RemoveAll` is not checked

(errcheck)